custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go

```diff
@@ -316,6 +318,17 @@ func (p *StackdriverProvider) ListAllMetrics() []provider.CustomMetricInfo {
 
 // GetExternalMetric queries Stackdriver for external metrics.
 func (p *StackdriverProvider) GetExternalMetric(ctx context.Context, namespace string, metricSelector labels.Selector, info provider.ExternalMetricInfo) (*external_metrics.ExternalMetricValueList, error) {
+	key := cacheKey{
+		namespace:      namespace,
+		metricSelector: metricSelector,
+		info:           info,
+	}
+	resp, ok := p.externalMetricCache.Get(key)
+
+	if ok {
```

Comment:
for additional telemetry, what do you think about adding some metrics to monitor how times we are getting cache hits?

(We can definitely do this in a future PR)

Reply:
Sounds good, can track this in a separate PR. Added a TODO for it (in the Get() method).